src/mbgl/style/source_impl.cpp

```diff
@@ -139,17 +139,23 @@ void Source::Impl::updateTiles(const UpdateParameters& parameters) {
     algorithm::updateRenderables(getTileFn, createTileFn, retainTileFn, renderTileFn,
                                  idealTiles, zoomRange, tileZoom);
 
+    bool isSameStrategy = (cacheStrategy == parameters.tileCacheStrategy);
     bool isSameDimension = (lastDimension == parameters.transformState.getSize());
     bool isSameZoomRange = (lastZoomRange == parameters.transformState.getZoomRange());
-    if (type != SourceType::Annotations && (cache.getSize() == 0 || !isSameDimension || !isSameZoomRange)) {
+    if (type != SourceType::Annotations && (!isSameStrategy || !isSameDimension || !isSameZoomRange)) {
+        cacheStrategy = parameters.tileCacheStrategy;
```

Comment:
The caching strategy can't change, since it's only ever set in the constructor. It shouldn't be part of the `UpdateParameters`.